app/models/user.rb

```diff
@@ -313,6 +313,35 @@ def compressed_export
     ActiveSupport::Gzip.compress Diaspora::Exporter.new(self).execute
   end
 
+  ######### Photos export ##################
+  mount_uploader :exported_photos_file, ExportedPhotos
+
+  def queue_export_photos
+    update exporting_photos: true
+    Workers::ExportPhotos.perform_async(id)
+  end
+
+  def perform_export_photos!
+    temp_zip = Tempfile.new([username, '_photos.zip'])
+    Zip::ZipOutputStream.open(temp_zip.path) do |zos|
+      photos.each do |photo|
+        begin
+          zos.put_next_entry(photo.remote_photo_name)
+          zos.print(photo.unprocessed_image.file.read)
+        rescue
```

Comment:
Do you have a more specific exception that you expect? Or even a better, a way to guard the code raising to from running in that case?